src/test/java/org/opensearch/knn/plugin/transport/UpdateModelGraveyardTransportActionTests.java

```diff
@@ -165,4 +188,201 @@ public void testCheckBlock() {
             .getInstance(UpdateModelGraveyardTransportAction.class);
         assertNull(updateModelGraveyardTransportAction.checkBlock(null, null));
     }
+
+    public void testGetIndicesUsingModel() throws IOException, ExecutionException, InterruptedException {
+        // Get update transport action
+        UpdateModelGraveyardTransportAction updateModelGraveyardTransportAction = node().injector()
+            .getInstance(UpdateModelGraveyardTransportAction.class);
+
+        String modelId = "test-model-id";
+        byte[] modelBlob = "testModel".getBytes();
+        int dimension = 2;
+
+        createIndex(MODEL_INDEX_NAME);
+
+        Model model = new Model(
+            new ModelMetadata(
+                KNNEngine.DEFAULT,
+                SpaceType.DEFAULT,
+                dimension,
+                ModelState.CREATED,
+                ZonedDateTime.now(ZoneOffset.UTC).toString(),
+                "",
+                "",
+                "",
+                MethodComponentContext.EMPTY
+            ),
+            modelBlob,
+            modelId
+        );
+
+        // created model and added it to index
+        addDoc(model);
+
+        // Create basic index (not using k-NN)
+        String testIndex1 = "test-index1";
+        createIndex(testIndex1);
+
+        // Attempt to add model id to graveyard with one non-knn index present, should succeed
```

Comment:
Not that you need to make a change but this single test is too big imo.
I would rather write a test of
1. create model
2. create 1 knn index not using the model, create 2 knn index using the model
3. verify adding model to graveyard which will fail with correct list of 2 knn index using the model
4. delete the 2 knn index using the model
5. verify adding model to graveyard will succeed.